packages/@aws-cdk/aws-vpcv2-alpha/lib/ipam.ts

```diff
@@ -0,0 +1,428 @@
+/* eslint-disable no-bitwise */
+import { CfnIPAM, CfnIPAMPool, CfnIPAMPoolCidr, CfnIPAMScope } from 'aws-cdk-lib/aws-ec2';
+import { IIpAddresses, VpcCidrOptions } from './vpc-v2';
+import { Construct } from 'constructs';
+import { CfnResource, Lazy, Names, Resource, Stack } from 'aws-cdk-lib';
+
+/**
+ * Represents the address family for IP addresses in an IPAM pool.
+ *
+ * @enum {string}
+ * @property {string} IP_V4 - Represents the IPv4 address family.
+ * @property {string} IP_V6 - Represents the IPv6 address family.
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html#cfn-ec2-ipampool-addressfamily
+ */
+export enum AddressFamily {
+  /**
+   * Represents the IPv4 address family.
+   * Allowed under public and private pool.
+   */
+  IP_V4 = 'ipv4',
+
+  /**
+   * Represents the IPv6 address family.
+   * Only allowed under public pool.
+   */
+  IP_V6 = 'ipv6',
+}
+
+/**
+ * Properties for creating an IPAM pool.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html
+ *
+ * @interface IpamPoolProps
+ * @extends {PoolOptions}
+ * @property {string} ipamScopeId - The ID of the IPAM scope.
+ */
+export interface IpamPoolProps extends PoolOptions {
+  /**
+   * Scope id where pool needs to be created
+   */
+  readonly ipamScopeId: string;
+
+  /**
+   * IPAM resource name
+   * @default autogenerated by CDK
+   */
+  readonly ipamPoolName?: string;
+
+}
+
+/**
+ * Defining the type for allowed AWS service values under IPAM pool option
+ */
+type AwsServiceName = 'ec2'
+
+/**
+ * Options for configuring an IPAM pool.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html
+ *
+ * @interface PoolOptions
+ * @property {AddressFamily} addressFamily - The address family of the pool (ipv4 or ipv6).
+ * @property {CfnIPAMPool.ProvisionedCidrProperty[]} [provisionedCidrs] - Information about the CIDRs provisioned to the pool.
+ * @property {string} [locale] - The locale (AWS Region) of the pool.
+ * @property {IpamPoolPublicIpSource} [publicIpSource] - The IP address source for pools in the public scope.
+ * @property {string} [awsService] - Limits which AWS service can use the pool.
+ */
+export interface PoolOptions{
+
+  /**
+   * addressFamily - The address family of the pool (ipv4 or ipv6).
+   */
+  readonly addressFamily: AddressFamily;
+
+  /**
+   * [provisionedCidrs] - Information about the CIDRs provisioned to the pool.
+   * @default - No CIDRs are provisioned
+   * @see CfnIPAMPool.ProvisionedCidrProperty
+   */
+  readonly provisionedCidrs?: CfnIPAMPool.ProvisionedCidrProperty[];
```

Comment:
so this is being passed to the underlying L1 though it is a single property it is defined as interface input 

```
    this.provisionedCidrs = props.provisionedCidrs;
  readonly provisionedCidrs?: Array<cdk.IResolvable | CfnIPAMPool.ProvisionedCidrProperty> | cdk.IResolvable;
```

this is the doc link for property https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html#cfn-ec2-ipampool-provisionedcidrs, it might be introduced as interface since there can be more properties to it, i can circle back with service team on this.